AirCasting/Utils/AveragingService.swift

```diff
@@ -81,39 +78,45 @@ final class AveragingService: NSObject, ObservableObject {
         }
     }
     
-    private func perform(sessionEntity: SessionEntity, averagingWindow: AveragingWindow) {
-        if let uuid = sessionEntity.uuid {
-            measurementStreamStorage.accessStorage { storage in
-                
-                let session = try? storage.getExistingSession(with: uuid)
+    private func perform(uuid: SessionUUID, averagingWindow: AveragingWindow, windowDidChange: Bool) {
+        measurementStreamStorage.accessStorage { storage in
+            let session = try? storage.getExistingSession(with: uuid)
+            session?.allStreams?.forEach { stream in
+                var averagedMeasurements: [MeasurementEntity] = (stream.allMeasurements ?? []).filter {
+                    $0.averagingWindow == averagingWindow.rawValue
+                }
                 
-                session?.allStreams?.forEach { stream in
-                    var averagedMeasurements: [MeasurementEntity] = (stream.allMeasurements ?? []).filter {
-                        $0.averagingWindow != AveragingWindow.zeroWindow.rawValue
-                    }
-                    let unaveragedMeasurements: [MeasurementEntity] = (stream.allMeasurements ?? []).filter {
-                        $0.averagingWindow == AveragingWindow.zeroWindow.rawValue
+                /// Step 1 - perform averaging again of elements in the previous threshold
+                if windowDidChange && averagingWindow == .secondThresholdWindow {
```

Comment:
Seems like we never enter this and the measurements from previous threshold still get wiped out

Reply:
Seems like this could work:
`let sessionDuration = Date().currentUTCTimeZoneDate.timeIntervalSince(session.startTime!)`